packages/data-service/src/data-service.ts

```diff
@@ -1513,6 +1534,51 @@ class DataServiceImpl extends WithLogContext implements DataService {
     return await coll.dropIndex(name);
   }
 
+  @op(mongoLogId(1_001_000_237))
+  async isListSearchIndexesSupported(ns: string): Promise<boolean> {
+    try {
+      await this.getSearchIndexes(ns);
+    } catch (err) {
+      return false;
+    }
+    return true;
+  }
+
+  @op(mongoLogId(1_001_000_238))
+  async getSearchIndexes(ns: string): Promise<SearchIndex[]> {
+    const coll = this._collection(ns, 'CRUD');
+    const cursor = coll.listSearchIndexes();
+    const indexes = await cursor.toArray();
+    void cursor.close();
+    return indexes as SearchIndex[];
+  }
```

Comment:
Did we drop support for cancellable feature here by removing `ExecutionOptions`?

Reply:
I removed it for now because we're not going to have a way for users to cancel fetching the indexes. We can add it if we ever need it.

Reply:
Maybe not for the users, but for us. 
As we are going to have polling, while the request is inflight and user navigates away, we probably want to abort it.
Same goes with when we are loading search indexes and user navigates away we also want to cancel this operation.

But feel free to merge it as its not needed now and we can come back to this later :)

Reply:
I think we can file a ticket and then triage it at project sync. Can't imagine us adding that functionality during milestone 1.